### ISSUE TITLE
Float conversion error in kubernetes_manifest

### ISSUE THREAD

**Issue by eczy-tempus** [hashicorp/terraform-provider-kubernetes#1659]
### Terraform Version, Provider Version and Kubernetes Version

```
Terraform version: Terraform 1.1.8
Kubernetes provider version: 2.10.0
Kubernetes version: 1.23
```

### Affected Resource(s)

- `kubernetes_manifest`

### Terraform Configuration Files
```hcl
resource "kubernetes_manifest" "manifest" {
  manifest = {
    "apiVersion" = "sloth.slok.dev/v1"
    "kind" = "PrometheusServiceLevel"
    "metadata" = {
      "name" = "namespace-service"
      "namespace" = "monitoring"
    }
    "spec" = {
      "service" = "namespace-service"
      "slos" = [
        {
          "alerting" = {
            "pageAlert" = {
              "disable" = true
            }
            "ticketAlert" = {
              "disable" = true
            }
          }
          "description" = "..."
          "name" = "namespace-service-slo"
          "objective" = 99.9
          "sli" = {
            "events" = {
              "errorQuery" = "..."
              "totalQuery" = "..."
            }
          }
        },
      ]
    }
  }
}
```

### Debug Output

<details>
  <summary>Truncated debug output</summary>

```
kubernetes_manifest.manifest: Refreshing state...
2022-03-24T15:35:12.295-0500 [ERROR] vertex "kubernetes_manifest.manifest" error: Failed encode 'object' attribute to Unstructured
2022-03-24T15:35:12.295-0500 [ERROR] vertex "kubernetes_manifest.manifest" error: Failed encode 'object' attribute to Unstructured
2022-03-24T15:35:12.295-0500 [ERROR] vertex "kubernetes_manifest.manifest (expand)" error: Failed encode 'object' attribute to Unstructured
╷
│ Error: Failed encode 'object' attribute to Unstructured
│ 
│   with kubernetes_manifest.manifest,
│   on main.tf line 10, in resource "kubernetes_manifest" "manifest":
│   10: resource "kubernetes_manifest" "manifest" {
│ 
│ AttributeName("spec"): [AttributeName("spec")]: cannot convert map element to
│ Unstructured: AttributeName("spec").AttributeName("slos"):
│ [AttributeName("spec").AttributeName("slos")]: cannot convert map element to
│ Unstructured: AttributeName("spec").AttributeName("slos").ElementKeyInt(0):
│ [AttributeName("spec").AttributeName("slos").ElementKeyInt(0)] cannot convert
│ list element to Unstructured:
│ AttributeName("spec").AttributeName("slos").ElementKeyInt(0).AttributeName("objective"):
│ [AttributeName("spec").AttributeName("slos").ElementKeyInt(0).AttributeName("objective")]:
│ cannot convert map element to Unstructured:
│ AttributeName("spec").AttributeName("slos").ElementKeyInt(0).AttributeName("objective"):
│ [AttributeName("spec").AttributeName("slos").ElementKeyInt(0).AttributeName("objective")]
│ inexact float approximation when converting number value
╵
2022-03-24T15:35:12.298-0500 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2022-03-24T15:35:12.308-0500 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/kubernetes/2.9.0/darwin_amd64/terraform-provider-kubernetes_v2.9.0_x5 pid=94216
2022-03-24T15:35:12.308-0500 [DEBUG] provider: plugin exited
```

</details>

### Steps to Reproduce

1. `terraform apply`

Note that running `terraform plan` will succeed, and the failure occurs after entering `yes` when prompted during `terraform apply`.

### Expected Behavior
What should have happened?

Terraform should apply this manifest to k8s successfully and create this custom resource within the `monitoring` namespace.

### Actual Behavior
What actually happened?

Terraform failed to apply the manifest due to floating point type conversion. This appears to have been thrown from [this](https://github.com/hashicorp/terraform-provider-kubernetes/blob/main/manifest/payload/from_value.go#L43) block.

```
 Error: Plugin error
│ 
│   with kubernetes_manifest.manifest,
│   on main.tf line 65, in resource "kubernetes_manifest" "manifest":
│   65: resource "kubernetes_manifest" "manifest" {
│ 
│ The plugin returned an unexpected error from
│ plugin.(*GRPCProvider).ApplyResourceChange: rpc error: code = Unknown desc =
│ AttributeName("spec"): [AttributeName("spec")]: cannot convert map element to
│ Unstructured: AttributeName("spec").AttributeName("slos"):
│ [AttributeName("spec").AttributeName("slos")]: cannot convert map element to
│ Unstructured: AttributeName("spec").AttributeName("slos").ElementKeyInt(0):
│ [AttributeName("spec").AttributeName("slos").ElementKeyInt(0)] cannot convert list
│ element to Unstructured:
│ AttributeName("spec").AttributeName("slos").ElementKeyInt(0).AttributeName("objective"):
│ [AttributeName("spec").AttributeName("slos").ElementKeyInt(0).AttributeName("objective")]:
│ cannot convert map element to Unstructured:
│ AttributeName("spec").AttributeName("slos").ElementKeyInt(0).AttributeName("objective"):
│ [AttributeName("spec").AttributeName("slos").ElementKeyInt(0).AttributeName("objective")]
│ inexact float approximation when converting number value
```

### Important Factoids

- We are using [Prometheus Operator](https://github.com/prometheus-operator/prometheus-operator) and [Sloth](https://github.com/slok/sloth) to generate these `PrometheusServiceLevel` resources via k8s CRDs.
- If you wrap the objective value in double quotes (making it a string), the initial `terraform apply` works, but then every subsequent `apply` or `plan` will not work since terraform will be unable to refresh its state due to this type conversion error.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by alexsomesan** [comment]
This is indeed an issue in the provider logic. The PR referenced above contains the fix.

**Comment by eczy-tempus** [comment]
After upgrading to Terraform 1.1.8, this issue still occurs. I've updated the original issue accordingly. Are there any updates on https://github.com/hashicorp/terraform-provider-kubernetes/pull/1661?

**Comment by alexsomesan** [comment]
The changes in #1661 are now available in provider version v2.11.0
